src/net/sourceforge/kolmafia/request/GenericRequest.java

```diff
@@ -2484,17 +2484,22 @@ private static void checkItemRedirection(final String location) {
     String nextAdventure = null;
 
     switch (itemId) {
-      case ItemPool.BLACK_PUDDING:
+      case ItemPool.BLACK_PUDDING -> {
         itemName = "Black Pudding";
         consumed = true;
-        break;
-
-      case ItemPool.DRUM_MACHINE:
+      }
+      case ItemPool.DRUM_MACHINE -> {
         itemName = "Drum Machine";
         consumed = true;
-        break;
-
-      case ItemPool.DOLPHIN_WHISTLE:
+      }
+      case ItemPool.APRIL_BAND_TOM -> {
+        if (InventoryManager.hasItem(ItemPool.WORM_RIDING_HOOKS)) {
+          return;
+        }
+        itemName = "Apriling Band Quad Tom";
+        consumed = true;
```

Comment:
does this still get hit given the url is `inventory.php?iid=11567&action=aprilplay` instead of straight use?

also I don't think the item itself is ever consumed

Reply:
Obviously I need to try it out myself and write a test.

I had no idea this was a thing until it was mentioned on the CyberRealm thread.

Reply:
I tested both with and without worm riding hooks in inventory and wrote tests for both.